lib/Intlc/Linter.hs

```diff
@@ -0,0 +1,37 @@
+module Intlc.Linter where
+
+import           Intlc.ICU
+import           Prelude   hiding (Type)
+
+data Status
+  = Success
+  | Failure Text
+  deriving (Eq, Show)
+
+countInterpolations :: NEStream -> Int
+countInterpolations = foldr go 0
+  where
+    count = \case
+      String      -> 0
+      Number      -> 0
+      Date {}     -> 0
+      Time {}     -> 0
+      PluralRef   -> 0
+      Bool {}     -> 1
+      Plural {}   -> 1
+      Select {}   -> 1
+      Callback {} -> 1
+    go :: Token -> Int -> Int
+    go t n = case t of
+      Plaintext {}              -> n
+      Interpolation ((Arg _ x)) -> count x + n
+
+lint :: Message -> Status
+lint m = case m of
+  Static {}        -> Success
```

Comment:
I'm guessing we'll need to make this composable when we're going to add more rules. e.g some rules may apply to `Static` message.